src/audio/volume/volume.h

```diff
@@ -36,15 +36,17 @@ struct sof_ipc_ctrl_value_chan;
 
 #if defined(__XCC__)
 # include <xtensa/config/core-isa.h>
-# if XCHAL_HAVE_HIFI4
-#  define VOLUME_HIFI4
-# elif XCHAL_HAVE_HIFI3
-#  define VOLUME_HIFI3
-# else
-#  define VOLUME_GENERIC
-# endif
+#if CONFIG_VOLUME_DEFAULT_OPT_LEVEL
+#if XCHAL_HAVE_HIFI4
+#  define CONFIG_VOLUME_HIFI4    1
+#elif XCHAL_HAVE_HIFI3
+#  define CONFIG_VOLUME_HIFI3    1
 #else
-# define VOLUME_GENERIC
+#  define CONFIG_VOLUME_GENERIC  1
+#endif
+#endif
+#else
+#  define CONFIG_VOLUME_GENERIC  1
```

Comment:
All this is NOT needed any more.

Reply:
@lgirdwood It is needed still, otherwise this PR cannot be submitted before we have defined a generic overlay and converted the github flows to use it.

Reply:
@kv2019i @btian1 the important part is to remove the dependence on xtos headers above and duplication of Kconfig, so this has to go.
@btian1 we will need for volume currently a
1) ```app/overlays/<platform>/cc_default.conf``` - uses default production values for xcc/xt-clang.
2) ```app/overlays/<platform>/gcc_default.conf``` - uses non HIFI values for xtensa GCC.
Once 1 & 2 are ready then we can plumb into the GH CI workflows.

Reply:
@btian1  @marc-hb  Discussed this with @lgirdwood  and one new idea. So starting premise is we want to get rid of the core-isa.h based logic in the source files and drive this logic from Kconfig). 

To make transition easier, what if we reverse the logic in volume.h and detect a Zephyr SDK build (plain gcc), and if detected, force VOLUME_GENERIC. 

This would allow to keep our (and Zephyr's) github workflows unmodified, but move to Kconfig-based selection of SIMD level for non-cloud builds.